### ISSUE TITLE
Number field types are not validated properly

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#773]
As reported in Slack, a field `field_volume_number` of type integer and value "Vol. 1, No. 2" passes `--check` but fails on ingest with a 422 `ERROR - Node for CSV record id:25024 not created, HTTP response code was 422, response body was b'{"message":"Unprocessable Entity: validation failed.\\nfield_volume_number.0.value: This value should be of the correct primitive type.\\n"}'`

**Comment by mjordan** [comment]
Need to check how Drupal expects integer and decimal/float values to be represented in JSON. If it's different than string text, a new field type in workbench_fields.py might be worthwhile.

**Comment by mjordan** [comment]
Number type determines the formatting of the JSON. Decimal field values are treated as strings (wrapped in quotes) but float and integer field values are not:

```
"field_issue_773_decimal": [
    {
      "value": "20.50"
    }
  ],
  "field_issue_773_float": [
    {
      "value": 30.9
    }
  ],
  "field_issue_773_integer": [
    {
      "value": 10
    }
```

**Comment by mjordan** [comment]
In field configs, decimal fields are field_type 'decimal', floats are field_type 'float', and integers are field_type 'integer'.

**Comment by mjordan** [comment]
issue_773 branch merged into main with 3f7b9662c510e2c5caaa4fb5b8bac2994180519b. If any problems with number-typed fields continue to exist, please post a comment here and I will reopen.
